lib/prawn/font/ttc.rb

```diff
@@ -0,0 +1,35 @@
+# font.rb : The Prawn font class
+#
+# Copyright November 2008, Jamis Buck. All Rights Reserved.
+#
+# This is free software. Please see the LICENSE and COPYING files for details.
+#
+require_relative 'ttf'
+
+module Prawn
+  class Font
+    # @private
+    class TTC < TTF
+      # Returns a list of the names of all named fonts in the given ttc file.
+      # They are returned in order of their appearance in the file.
+      #
+      def self.fonts(file)
+        TTFunk::Collection.open(file) do |f|
+          list = []
+
+          f.each do |font|
+            list << font.name.font_name.join(', ')
+          end
+
+          list
+        end
+      end
+
+      private
+
+      def read_ttf_file
+        TTFunk::File.from_ttc(@name, @options[:font] || 0)
```

Comment:
Yeah, `:font` is not the best name when the value is an index, but that's how dfont.rb originally defined it. The difference is that dfont.rb supports specifying fonts by both name and index. I'll rework this so that it supports name look ups as well, to mimic the dfont implementation.